build.cake

```diff
@@ -354,6 +354,72 @@ Task("PackageZip")
         Zip(CurrentImageDir, File(ZIP_PACKAGE), zipFiles);
     });
 
+Task("CreateToolManifest")
+    .Does(() =>
+    {
+        var result = StartProcess("dotnet.exe", new ProcessSettings {  Arguments = "new tool-manifest --force" });
+    });
+
+Task("InstallSigningTool")
+    .Description("Installs the signing tool")
+    .IsDependentOn("CreateToolManifest")
+    .Does(() =>
+    {
+        var result = StartProcess("dotnet.exe", new ProcessSettings {  Arguments = "tool install SignClient" });
+    });
+
+Task("SignPackages")
+    .Description("Signs the NuGet packages")
+    .IsDependentOn("InstallSigningTool")
+    //.IsDependentOn("PackageFramework")
+    .Does(() =>
+    {
+        // Get the secret.
+        var secret = EnvironmentVariable("SIGNING_SECRET");
+        if(string.IsNullOrWhiteSpace(secret)) {
+            throw new InvalidOperationException("Could not resolve signing secret.");
+        }
+        Information($"Signing secret: {secret}");
+
+        // Get the user.
+        var user = EnvironmentVariable("SIGNING_USER");
+        if(string.IsNullOrWhiteSpace(user)) {
+            throw new InvalidOperationException("Could not resolve signing user.");
+        }
+        Information($"Signing user: {user}");
+
+        var signClientPath = Context.Tools.Resolve("SignClient.exe") ?? Context.Tools.Resolve("SignClient") ?? throw new Exception("Failed to locate sign tool");
```

Comment:
Maybe DotNetCoreTool would be better than StartProcess for this too?

Reply:
When I first did this for the console, the Cake build team helped me with this and pointed me at their signing, https://github.com/cake-build/cake/blob/develop/build.cake#L415

Reply:
Alright, that is helpful. Sounds good.